aten/src/ATen/native/Convolution.cpp

```diff
@@ -652,6 +652,88 @@ static at::Tensor subtensor(at::Tensor& tensor, int dim, int groups, int g) {
   return tensor.narrow(dim, n * g, n).contiguous();
 }
 
+namespace {
+
+std::pair<Tensor, Tensor> complex_to_real(const Tensor& inp) {
+  auto inp_view_as_complex = at::view_as_real(inp);
+  auto dim_i = inp_view_as_complex.dim() - 1;
+  auto i_r = inp_view_as_complex.select(dim_i, 0);
+  auto i_i = inp_view_as_complex.select(dim_i, 1);
+  return std::make_pair(i_r, i_i);
+}
+
+at::Tensor complex_convolution(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    IntArrayRef output_padding,
+    int64_t groups) {
+  check_input_same_type_as_parameters(input, weight, bias);
+  Tensor i_r, i_i, w_r, w_i;
+  std::tie(i_r, i_i) = complex_to_real(input.resolve_conj());
```

Comment:
`resolve_conj` is not needed since the conjugation will be resolved by the fallback kernel before reaching here

Reply:
Without `resolve_conj`, these OpInfo tests failed.

<details>

```
_________________________________________________ TestMathBitsCPU.test_conj_view_nn_functional_conv1d_cpu_complex64 __________________________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 831, in test_conj_view
    self._test_math_view(device, dtype, op, samples, math_op_physical, math_op_view, is_bit_set, torch.is_complex)
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 781, in _test_math_view
    forward_with_mathview = op(cloned1, *sample.args, **sample.kwargs)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/common_methods_invocations.py", line 824, in __call__
    return self.op(*args, **kwargs)
RuntimeError: view_as_real doesn't work on unresolved conjugated tensors.  To resolve the conjugate tensor so you can view it as real, use self.resolve_conj(); however, be warned that the resulting tensor will NOT alias the original.
_______________________________________________ TestMathBitsCPU.test_neg_conj_view_nn_functional_conv1d_cpu_complex128 _______________________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 864, in test_neg_conj_view
    self._test_math_view(device, dtype, op, samples, math_op_physical, math_op_view, is_bit_set,
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 781, in _test_math_view
    forward_with_mathview = op(cloned1, *sample.args, **sample.kwargs)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/common_methods_invocations.py", line 824, in __call__
    return self.op(*args, **kwargs)
RuntimeError: view_as_real doesn't work on unresolved conjugated tensors.  To resolve the conjugate tensor so you can view it as real, use self.resolve_conj(); however, be warned that the resulting tensor will NOT alias the original.
________________________________________________ TestMathBitsCUDA.test_conj_view_nn_functional_conv1d_cuda_complex64 _________________________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 831, in test_conj_view
    self._test_math_view(device, dtype, op, samples, math_op_physical, math_op_view, is_bit_set, torch.is_complex)
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 781, in _test_math_view
    forward_with_mathview = op(cloned1, *sample.args, **sample.kwargs)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/common_methods_invocations.py", line 824, in __call__
    return self.op(*args, **kwargs)
RuntimeError: view_as_real doesn't work on unresolved conjugated tensors.  To resolve the conjugate tensor so you can view it as real, use self.resolve_conj(); however, be warned that the resulting tensor will NOT alias the original.
______________________________________________ TestMathBitsCUDA.test_neg_conj_view_nn_functional_conv1d_cuda_complex128 ______________________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 864, in test_neg_conj_view
    self._test_math_view(device, dtype, op, samples, math_op_physical, math_op_view, is_bit_set,
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/test_ops.py", line 781, in _test_math_view
    forward_with_mathview = op(cloned1, *sample.args, **sample.kwargs)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/common_methods_invocations.py", line 824, in __call__
    return self.op(*args, **kwargs)
RuntimeError: view_as_real doesn't work on unresolved conjugated tensors.  To resolve the conjugate tensor so you can view it as real, use self.resolve_conj(); however, be warned that the resulting tensor will NOT alias the original.
```

</details>

Reply:
I think since after calling `conj` in test. The immediate next thing the tensor hits via `conv1d` (which is composite) is `view_as_real`. The conj stays unresolved.

 I think script below traces similar path and leads to same error.
```python
import torch
t = torch.randn(3, dtype=torch.complex64)
z = torch.view_as_real(t.conj())
```

Reply:
ahh gotcha makes sense